model/buildlogger.go

```diff
@@ -21,6 +21,16 @@ import (
 
 const buildloggerCollection = "buildlogs"
 
+func CreateLog(info LogInfo, artifact LogArtifactInfo) *Log {
+	return &Log{
+		ID:       info.ID(),
+		Info:     info,
+		Artifact: artifact,
```

Comment:
aren't we going to want to support having more than one artifact?

Reply:
Not in my design, a log in the database maps to one sub-bucket in aws (where the log is broken up into chunks)

Reply:
how will we know how many subchunks there are supposed to be, and how many lines are stored in each chunk? seems like we'd want to be able to have that info per artifact.

Reply:
We can add something to store the number of chunks (and maybe the number of lines in each chunk) but that would still be one artifact. Storing the number of chunks would just be for sanity/error checking, i.e. making sure when we download the log file that we have the right number of chunks and each chunk has the right number of lines. Since the timestamp is the key of each chunk file, the entire sub-bucket is downloaded and the log is stitched back together by sorting the filenames.